src/lib/getPriorityIndex.ts

```diff
@@ -299,29 +351,112 @@ const priorityGroups: readonly PriorityGroup[] = [
       "objectFit",
       "objectPosition",
       "float",
-      "fill",
-      "stroke",
       "outline",
     ],
+    priority: stylePropsPriority["Other Style Props"],
   },
 ];
 
-export function getPriorityIndex(key: string): number {
-  const index = priorityGroups.findIndex((group) => {
+export function getPriority(key: string, config: Config): number {
+  // getPriority returns a number. The smaller is the heigher priority.
+  // Properties will have their "group priority", determined by which group property belongs, and "inGroup priority", determined by index in that group.
+  const { firstProps, lastProps, componentSpecificProps } = config;
+  const indexInFirstProps = firstProps.indexOf(key);
+  const indexInLastProps = lastProps.indexOf(key);
+
+  if (indexInFirstProps !== -1) {
+    return calcPriorityFromIndex({ type: "reservedFirstProps", value: indexInFirstProps }, config);
+  }
+  if (indexInLastProps !== -1) {
+    return calcPriorityFromIndex({ type: "reservedLastProps", value: indexInLastProps }, config);
+  }
+
+  if (componentSpecificProps) {
+    const index = componentSpecificProps.indexOf(key);
+    if (index !== -1) {
+      return calcPriorityFromIndex({ type: "componentSpecificProps", value: index }, config);
+    }
+  }
+
+  // Then it can be either `stylePropsPriority` or `otherPropsPriority`
+  const groupIndex = priorityGroups.findIndex((group) => {
     return group.keys.includes(key);
   });
-  return index === -1 ? Number.MAX_SAFE_INTEGER : index;
+
+  const isStyleProps = groupIndex > -1;
+  if (isStyleProps) {
+    const keyIndex = getIndexInGroup(key, groupIndex);
+    return calcPriorityFromIndex({ type: "styleProps", groupIndex, keyIndex }, config);
+  }
+
+  return calcPriorityFromIndex({ type: "otherProps" }, config);
 }
 
-export const priorityGroupsLength = priorityGroups.length;
+type Index =
+  | { type: "reservedFirstProps" | "componentSpecificProps" | "reservedLastProps"; value: number }
+  | { type: "styleProps"; groupIndex: number; keyIndex: number }
+  | { type: "otherProps" };
 
-export const getIndexInPriority = (key: string, groupIndex: number): number => {
-  const keys = priorityGroups[groupIndex].keys;
-  if (keys) {
-    const index = keys.indexOf(key);
-    return index === -1 ? Number.MAX_SAFE_INTEGER : index;
+const calcPriorityFromIndex = (index: Index, config: Config) => {
```

Comment:
There are two reasons why we are passing `index` as an object
1. for the possibility that the logic may become more complex in the future
2. because TypeScript's overload is not as strong in type inference as Discriminated Union.

ref: https://d.potato4d.me/entry/20200831-factory-args/